client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
 
-To create a flow, you only have to implement `initialize` and `useStepNavigation`. `initialize` can do any checks you need and it should finally return an array of step objects, `useStepNavigation` is the engine where you make navigation decisions. This hook returns an object of type [`NavigationControls`](./declarative-flow/internals/types.ts):
+```ts
+async function initialize() {
+	const hasAccessToFlow = await isAgencyUser();
+
+	if ( ! hasAccessToFlow ) {
+		window.location = '/home';
+		// Return false to halt the fetching and execution of the steps. Making the redirect much faster.
+		return false;
+	}
+
+	return [ STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+}
+```
+
+### Notes
+
+1. A successful flow is a rare event, so most flows are sadly ephemeral. Please keep as much logic, CSS, and code as possible into the flow folder itself. So that when it's deleted, clean up is easy.
+2. Adding testing steps in your flow's README goes a long way. It would be really appreciated.
+3. Feel free to ask for help any time. You can post in `#dotcom-stepper` or ping `@alshakero`.
+4. Please try to avoid modifying the framework's code around your flow. It should be the very last resort.
+
+### Deleting your flow
+
+After deleting your flow, please setup a redirect for it [here](/client/landing/stepper/utils/flow-redirect-handler.ts), this redirect users who land on it to a happy path.
+
+### Making a Step
+
+Note: Before making a step, please make sure there isn't already a suitable step in [`steps.tsx`](/client/landing/stepper/declarative-flow/internals/steps.tsx) file. If you do make a step you'll have to add it to that file.
 
-There is also an optional `useSideEffect` hook. You can implement this hook to run any side-effects to the flow. You can prefetch information, send track events when something changes, etc...
+**Note**: Please make sure that your step has a unique slug.
 
-There is a required `isSignupFlow` flag that _MUST be `true` for signup flows_ (generally where a new site may be created), and should be `false` for other flows. The `isSignupFlow` flag controls whether we'll trigger a `calypso_signup_start` Tracks event when the flow starts. For signup flows, you can also supply additional event props to the `calypso_signup_start` event by implementing the optional `useTracksEventProps()` hook on the flow.
+**A step is simply a React component that**:
+
+1. Renders UI that collects user input.
+2. Submit said input to the flow.
+3. May or may not accept props from the flow.
+
+#### Code example
 
 ```tsx
 /**
- * This is the return type of useStepNavigation hook
+ * Each step must be typed as Step. And it should declare the types of the data it submits and the data it accepts.
  */
-export type NavigationControls = {
+const SelectImportedSiteSource: Step< {
+	// This steps submits `platform` and `url` strings.
+	submits: {
+		platform: 'Wix' | 'Squarespace';
```

Comment:
It is a little weird to see these names here. 😅 

Could we do something like "WordPress" and "Pressable" just for the sake of the example?

(Same comment for the `select` options below.)

Reply:
hehe I thought it was a nice quip 🤣 

Will do.